pkg/controller/common/annotation/controller_version.go

```diff
@@ -15,7 +15,11 @@ import (
 )
 
 // ControllerVersionAnnotation is the annotation name that indicates the last controller version to update a resource
-const ControllerVersionAnnotation = "common.k8s.elastic.co/controller-version"
+const (
+	ControllerVersionAnnotation       = "common.k8s.elastic.co/controller-version"
+	MinCompatibleControllerVersion    = "0.10.0-SNAPSHOT"
```

Comment:
These names are confusing to me at least, just going by the names it's not clear why we would need both the last incompatible version and the minimum compatible version. Going by how they're used I think it's because we're using "last incompatible" to mean the version before we started adding the annotation, but it might be worth finding more descriptive names and/or adding explanatory comments on why these exist

Reply:
Will that work? I think it needs to be a parseable version for later versions of the controller to know they shouldn't do anything with it when it does a version comparison, but definitely possible I'm misunderstanding

Reply:
oh, indeed! Then maybe:

```
UnknownControllerVersion = "0.0.0-UNKNOWN" // may match resources created with ECK-0.8.0
```